apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.ejs

```diff
@@ -333,7 +373,13 @@
             <% log.forEach(function(row, index) { %>
             <tr>
               <td class="text-nowrap"><%= row.formatted_date %></td>
-              <td><%= row.auth_user_uid %></td>
+              <td>
+                <% if (row.auth_user_uid) { %>
+                  <%= row.auth_user_uid %>
+                <% } else { %>
+                  &mdash;
+                <% } %>
+              </td>
```

Comment:
I made this change to do something sensible for historical data (that is, variants where `broken_at IS NOT NULL` but `broken_by IS NULL`.

Reply:
Should we instead backfill `variants.broken_by` with the `user_id` of the owner of the variant? I think that would be correct for all currently-existing variants. This would mean that we don't have to allow `auth_user_uid` to be nullable in the log data, which feels like a nice property to retain?

This is not critical, of course, and we could always do it as a later followup.

Reply:
Thinking about this further: do we want to differentiate between "this variant was explicitly broken by a user" and "this variant was marked as broken by the system"? Even after this PR, variants that are marked as "broken" because they failed to generate will have `broken_by IS NULL`.

Reply:
Even after this PR, variants that are broken "by the system" (those that fail to generate/prepare) will also have a `broken_by IS NULL`. Is that something you'd like to see change?

Reply:
Not really, if we aren't going to backfill (which is ok by me) then I think it's fine as is. My original comment was just triggered by an instinctive desire for NOT NULL, but it isn't so important in this case.